server/app/views/admin/apikeys/ApiKeyNewOneView.java

```diff
@@ -0,0 +1,142 @@
+package views.admin.apikeys;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.p;
+
+import annotations.BindingAnnotations.EnUs;
+import com.github.slugify.Slugify;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import controllers.admin.routes;
+import j2html.tags.ContainerTag;
+import java.util.Optional;
+import play.data.DynamicForm;
+import play.i18n.Messages;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.apikey.ApiKeyService;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.components.FieldWithLabel;
+import views.style.Styles;
+
+/** Renders a page for adding a new ApiKey. */
+public final class ApiKeyNewOneView extends BaseHtmlView {
+  private final AdminLayout layout;
+  private final Messages enUsMessages;
+  private final Slugify slugifier = new Slugify();
+
+  @Inject
+  public ApiKeyNewOneView(AdminLayout layout, @EnUs Messages enUsMessages) {
+    this.layout = checkNotNull(layout);
+    this.enUsMessages = checkNotNull(enUsMessages);
+  }
+
+  public Content render(Request request, ImmutableSet<String> programNames) {
+    return render(request, programNames, /* dynamicForm= */ Optional.empty());
+  }
+
+  public Content render(
+      Request request, ImmutableSet<String> programNames, Optional<DynamicForm> dynamicForm) {
+    String title = "Create a new API key";
+
+    ContainerTag formTag =
+        form()
+            .withMethod("POST")
+            .with(
+                makeCsrfTokenInputTag(request),
+                setStateIfPresent(
+                        FieldWithLabel.input()
+                            .setFieldName(ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                            .setId(ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                            .setLabelText("API key name"),
+                        dynamicForm,
+                        ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                    .getContainer(),
+                h2("Expiration date"),
+                p(
+                    "Specify a date when this API key will no longer be valid. The expiration date"
```

Comment:
likewise to the other html page, making this a const in the file will likely aid readability all around.

Reply:
done